### PR TITLE
flush out ExecStatus structures when a new service cycle begins

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/core/execute.h [hunk 1]
```diff
@@ -130,6 +135,9 @@ typedef struct ExecDirectory {
         mode_t mode;
 } ExecDirectory;
 
+/* Enocdes configuration parameters applied to invoked commands. Does not carry runtime data, but only configuration
```
Comment (dnicolodi) [inline src/core/execute.h]: `s/Enocdes/Encodes/`

File: src/core/execute.h [hunk 2]
```diff
@@ -130,6 +135,9 @@ typedef struct ExecDirectory {
...
+   changes sourced from unit files and suchlike. ExecContext objects are usually embedded into Unit objects, and do not
+   change after loaded once. */
```
Comment (dnicolodi) [inline src/core/execute.h]: `s/after loaded once/after being loaded once/` probably `once` can be dropped.

**Comment by poettering** [comment]
force pushed a new version with the two issues @dnicolodi discovered fixed! ptal!

File: src/core/execute.c [hunk 3]
```diff
@@ -4342,26 +4342,30 @@ void exec_context_free_log_extra_fields(ExecContext *c) {
...
 void exec_status_exit(ExecStatus *s, const ExecContext *context, pid_t pid, int code, int status) {
         assert(s);
 
-        if (s->pid && s->pid != pid)
-                zero(*s);
+        if (s->pid != pid) {
+                *s = (ExecStatus) {
+                        .pid = pid,
+                };
+        }
```
Comment (keszybz) [inline src/core/execute.c:4370]: External braces could be dropped.

File: src/core/execute.h [hunk 4]
```diff
@@ -100,13 +102,16 @@ struct ExecCommand {
         LIST_FIELDS(ExecCommand, command); /* useful for chaining commands */
 };
 
+/* Encapsulates certain aspects of the runtime environment that is to be shared between multiple otherwise separate
+ * invocations of commands. Specifically, this allows sharing of /tmp and /var/tmp data as well as network namespaces
+ * between invocations of commands. This is a reference counted object, with one reference taken by each currently
+ * active command invocation that wants to share this runtime. */
```
Comment (keszybz) [inline src/core/execute.h:108]: Nice.

**Pull request merged**